File: Rearrange_Array_Elements_By_Sign_1.cpp
/*
Link of the question : https://leetcode.com/problems/rearrange-array-elements-by-sign/
Leetcode question number : 2149
*/
class Solution
{
public:
    vector<int> rearrangeArray(vector<int> &nums)
    {
        int n = nums.size();
        vector<int> ans(n);
        int a = 0, b = 1;
        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] > 0)
            {
                ans[a] = nums[i];
                a = a + 2;
            }
            else
            {
                ans[b] = nums[i];
                b = b + 2;
            }
        }
        return ans;
    }
};

File: Rearrange_Array_Elements_By_Sign_1.py
'''
Link of the question : https://leetcode.com/problems/rearrange-array-elements-by-sign/
Leetcode question number : 2149
'''


class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        positives = []
        negatives = []
        for x in nums:
            if x < 0:
                negatives.append(x)
            else:
                positives.append(x)
        ans = []
        for a, b in zip(positives, negatives):
            ans.append(a)
            ans.append(b)
        return ans
